packages/flutter_tools/lib/src/ios/ios_workflow.dart

```diff
@@ -32,65 +33,64 @@ class IOSWorkflow extends DoctorValidator implements Workflow {
 
   bool get hasIDeviceId => exitsHappy(<String>['idevice_id', '-h']);
 
-  bool get hasWorkingLibimobiledevice {
+  Future<bool> get hasWorkingLibimobiledevice async {
     // Verify that libimobiledevice tools are installed.
     if (!hasIDeviceId)
       return false;
 
     // If a device is attached, verify that we can get its name.
-    final String result = runSync(<String>['idevice_id', '-l']);
-    if (result.isNotEmpty && !exitsHappy(<String>['idevicename']))
+    final ProcessResult result = (await runAsync(<String>['idevice_id', '-l'])).processResult;
+    if (result.exitCode == 0 && result.stdout.isNotEmpty && !await exitsHappyAsync(<String>['idevicename']))
       return false;
 
     return true;
   }
 
-  bool get hasIDeviceInstaller => exitsHappy(<String>['ideviceinstaller', '-h']);
+  Future<bool> get hasIDeviceInstaller => exitsHappyAsync(<String>['ideviceinstaller', '-h']);
```

Comment:
The non-async version is still used elsewhere. @tvolkert is the plan to kill the sync version?